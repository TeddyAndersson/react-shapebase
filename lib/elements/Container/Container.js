import React from 'react';
import _ from 'lodash';
import PropTypes from 'prop-types';
import styled, {css} from 'styled-components';
import Grid from '../Grid'

const Element = (props) => {
  const {
    children,
    className,
    size,
  } = props;

  return (
    <Grid columns={1}>
        <Grid.Item columnStart={1} columnEnd={2}>
            <div className={className}>
                {children}
            </div>
        </Grid.Item>
    </Grid>
  );
};

Element.propTypes = {
  /**
    * Primary content.
    */
  children: PropTypes.node,

  /**
    * Additonal classes.
    */
  className: PropTypes.string,

  /**
    * Determine if the container shoudl be fluid.
    */
  fluid: PropTypes.bool,

  /**
    * Determine the text alignment in the container.
    */
  textAlign: PropTypes.oneOf(['center', 'left', 'right', 'justify']),

};

Element.defaultProps = {
    textAlign: 'left',
    fluid: false,
}


/** Styling the Component with styled-components */
const Container = styled(Element)`
    margin-top: 0;
    margin-bottom: 0;
    display: block;
    max-width: 40em;
    width: auto;
    text-align: ${props => props.textAlign};

    ${ props => props.fluid && css`
        max-width: none;
    `}
`;


export default Container;
