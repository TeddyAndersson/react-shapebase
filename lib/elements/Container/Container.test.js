import React from 'react';
import Container from './Container'
import renderer from 'react-test-renderer';

test('should render Container', () => {
  const component = renderer.create(
    <Container textAlign="left">
      A container with it's text left aligned
    </Container>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})
