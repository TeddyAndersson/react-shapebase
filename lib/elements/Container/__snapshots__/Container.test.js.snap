// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Container 1`] = `
<div
  className="sc-bwzfXH faVpcD"
>
  <div
    className="sc-bdVaJa gdNUZi"
  >
    <div
      className="sc-htpNat eUMCnr"
    >
      A container with it's text left aligned
    </div>
  </div>
</div>
`;
